s3/replication/common/src/s3replicationcommon/job.py

```diff
@@ -68,9 +68,11 @@ def __init__(self, obj):
             self._obj = obj
         else:
             self._obj = {}
+
         # There are 2 identifiers, job_id which is generated
         # and replication id that is sent by job creator.
-        self._id = uuid.uuid4()
+        self._id = str(uuid.uuid4())
+        self._obj['job_id'] = self._id
```

Comment:
you will also need below in `def from_json(self, json_string):` 

```
if self._obj.get("job_id") is None:
            # Job id not present in json_string, use the generated one
            self._obj["job_id"] = self._id
        else:
            # json_string already had job id use it.
            self._id = self._obj["job_id"]
```

Reply:
done